sklearn/metrics/_scorer.py

```diff
@@ -651,6 +654,13 @@ def _get_response_method(response_method, needs_threshold, needs_proba):
     return response_method
 
 
+def _get_response_method_name(response_method):
+    if isinstance(response_method, partial):
+        return _get_response_method_name(response_method.func)
+
+    return response_method.__name__
```

Comment:
```suggestion
def _get_response_method_name(response_method):
    try:
        return response_method.__name__
    except AttributeError:
        return _get_response_method_name(response_method.func)
```

This might make more sense, since users can manually set `__name__` on partials and we should then use that one.